### PR TITLE
feat(orders): add NIP-17 order message boundary

### LINKED ISSUE #1084: [BUG] Message not viewable to sender after sending follow up message after an order to the seller
**Describe the bug**
After making an order, I do not see the messages to the seller in other apps. I originally thought that this was due to using NIP-17 gift wraps that weren't supported by other clients, but I don't see it in the Lightning Piggy mobile app which does support NIP-17. Looking at the logs, I then wondered if you are sending the message via the preferred relays of the sender and/or receiver or just the plebeian relay?

**To Reproduce**

1. Go to buy the following sticker pack: https://plebeian.market/products/4866bd0f50b6828b2b7e3ef048789f8ff59b9daae55b8539de5769e8f9b3ceef
2. Proceed to place the order (using Amber and WebLN to make the payment)
4. Confirm if the recipient received the message

**Expected behavior**
I should be able to see the order messages in other messaging clients.

**Screenshots**

Image

- Browser: Edge
- OS: Kubuntu

**Additional context**
- I signed each event using Amber.
- I used WebLN to make the payment (successfully)
- I sent a direct message using Primal (NIP-04) and Lightning Piggy (NIP-17) and can see those two messages, but not the ORder Placed or Payment Receipt.

### PR DESCRIPTION
**Pull request by maximotodev**
## Summary

Adds a PR-1 foundation for #1084 by introducing a tested NIP-17/Gamma order-message boundary without wiring it into checkout, payment publishing, reads, relay resolution, or global NDK publish behavior.

## What changed

- Added ADR-013 for NIP-17 order message transport.
- Added `src/lib/nostr/nip17.ts` for recipient gift wrap + sender self-wrap creation.
- Added `src/lib/orders/orderMessageRumor.ts` for pure Gamma-shaped order-message rumor builders and schema-backed validation.
- Added covered unit tests for:
  - recipient and sender self-wraps
  - canonical unsigned inner rumors
  - no public gift-wrap leakage of order/payment/PII sentinels
  - NIP-44 failure behavior
  - NIP-17 timestamp randomization bounds
  - kind 14, kind 16 type 1–4, and kind 17 inner rumor shapes

## Non-goals

- No checkout/order publish wiring.
- No payment publish wiring.
- No read-path migration.
- No relay resolver or kind 10050 routing implementation.
- No global NDK publish or relay behavior changes.
- No removal of legacy raw kind 14/16/17 reads.

## Validation

- `bun test src/lib/__tests__/nip17.test.ts`
- `bun test src/lib/__tests__/orderMessageRumor.test.ts`
- `bun run test:unit`

Latest local result:

```text
109 pass
0 fail
425 expect() calls
```

### CODE REVIEW

**Comment by maximotodev** [comment]
@codex review

File: src/lib/orders/orderMessageRumor.ts [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+	content?: string
+	createdAt?: number
+}
+
+export type CreatePaymentRequestRumorParams = {
+	merchantPubkey: string
+	buyerPubkey: string
+	orderId: string
+	amountSats: number
+	paymentMethods: PaymentMethodInput[]
```
[the remark this answers is not in the document]
Comment (maximotodev) [reply src/lib/orders/orderMessageRumor.ts]: Good catch. I patched the boundary to fail fast for empty `paymentMethods` instead of accepting a shape that cannot satisfy `PaymentRequestSchema`.

This keeps PR 1 schema-compatible. Future publish wiring will need to handle the pending-payment case before wrapping, preserving the existing workflow where missing seller/V4V payment details can remain pending instead of being silently dropped.

**Comment by maximotodev** [comment]
@codex review

File: src/lib/orders/orderMessageRumor.ts [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+		pubkey: params.buyerPubkey,
+		created_at: params.createdAt ?? unixNow(),
+		content: params.content ?? 'Order created',
+		tags,
+	})
+}
+
+export function createPaymentRequestRumor(params: CreatePaymentRequestRumorParams): OrderMessageRumor {
+	if (params.paymentMethods.length === 0) {
+		throw new Error('Payment request rumors require at least one payment method')
```
[the remark this answers is not in the document]
Comment (maximotodev) [reply src/lib/orders/orderMessageRumor.ts]: I aligned the boundary with the existing pending-payment workflow and Gamma’s manual payment request shape.

The patch now allows `paymentMethods: []`, keeps the generated kind 16/type 2 rumor canonical, and loosens `PaymentRequestSchema` so `payment` remains optional while `p`, `subject`, `type`, `order`, and `amount` stay required. Added regression coverage for the pending-payment case.

**Review by c03rad0r — APPROVED** [review]
Solid foundation PR — approving. The scope discipline is exactly right: a tested NIP-17/Gamma boundary that doesn't touch checkout, payment wiring, read paths, or global NDK behavior. The PII sentinel tests (verifying that encrypted fields don't leak into public gift-wrap tags/content) are a nice security-conscious touch.

A few non-blocking notes:

**1. Both gift wraps share the same `createdAt` timestamp.** `createNip17GiftWrapsWithSigner` computes `createdAt` once (line 926) and passes it to both the recipient wrap and sender self-wrap. In strict NIP-17, each gift wrap could have its own independently randomized timestamp. Not a spec violation for a foundation PR — but worth noting for PR 3 (publish wiring) where per-wrap randomization would better obscure timing correlation.

**2. `assertOrderMessageRumor` swallows the actual validation error.** The catch block on line 1231 throws a generic `"Invalid order message rumor"` regardless of the underlying cause (missing pubkey, non-canonical id, signed rumor, Zod parse failure). For debugging in development and integration tests, consider preserving the original error message — e.g., `throw new Error(\`Invalid order message rumor: ${e.message}\`)` or at least logging it. Not a concern for the security boundary (fail-closed is correct), just debuggability.

**3. Missing negative tests for the assertion function.** The happy paths and the "missing `p` tag" case are covered, but these assertion branches aren't directly tested:
- Rumor with `sig` present (should reject: "rumor must be unsigned")
- Rumor with non-canonical `id` (should reject: "non-canonical rumor id")
- Rumor with unknown `kind` (should reject: "unsupported order message kind")

These are edge cases in `assertOrderMessageRumor` that would benefit from explicit negative tests to lock in the contract.

**4. Schema change: `payment` tag now optional in `PaymentRequestSchema`.** This is backwards-compatible (relaxing validation) and the comment explains the rationale (pending/manual payment setup). The test for "pending payment request without payment methods" covers it. Just flagging that existing code that assumes `payment` tags are always present in validated payment requests should be audited when this gets wired into the read path (PR 4).

**What's done well:**
- Correct NIP-59 reuse — `createNip59GiftWrapWithSigner` is called for both recipient and sender wraps, with a rumor-id consistency check (line 944) to catch any drift.
- Canonical id computation via `getEventHash` ensures rumors are tamper-evident.
- All 6 Gamma message types (kind 14, kind 16 types 1-4, kind 17) have dedicated builders with proper tag shapes and Zod schema validation.
- The mock signer in tests properly implements the full `NDKSigner` interface and can simulate NIP-44 unavailability — the "fails closed" test is important.
- ADR-013 is thorough and well-structured with clear PR sequencing, alternatives considered, and open questions documented.
